homeassistant/components/iotty/cover.py

```diff
@@ -0,0 +1,218 @@
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return self._iotty_device.percentage
+
+    @property
+    def is_closed(self) -> bool:
+        """Return true if the Shutter is closed."""
+        _LOGGER.debug(
+            "Retrieve device status for %s ? %s : %s",
+            self._iotty_device.device_id,
+            self._iotty_device.status,
+            self._iotty_device.percentage,
+        )
+        return (
+            self._iotty_device.status == ShutterState.STATIONARY
+            and self._iotty_device.percentage == 0
+        )
+
+    @property
+    def is_opening(self) -> bool:
+        """Return true if the Shutter is opening."""
+        return self._iotty_device.status == ShutterState.OPENING
+
+    @property
+    def is_closing(self) -> bool:
+        """Return true if the Shutter is closing."""
+        return self._iotty_device.status == ShutterState.CLOSING
+
+    @property
+    def supported_features(self) -> CoverEntityFeature:
+        """Flag supported features."""
+        return CoverEntityFeature(0) | (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.STOP
+            | CoverEntityFeature.SET_POSITION
+        )
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id, self._iotty_device.cmd_open()
+        )
+        await self.coordinator.async_request_refresh()
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close cover."""
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id, self._iotty_device.cmd_close()
+        )
+        await self.coordinator.async_request_refresh()
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the cover to a specific position."""
+        percentage = kwargs[ATTR_POSITION]
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id,
+            self._iotty_device.cmd_move_to(),
+            {"open_percentage": percentage},
+        )
+        await self.coordinator.async_request_refresh()
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id, self._iotty_device.cmd_stop()
+        )
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+
+        device: Device = next(
+            device
+            for device in self.coordinator.data.devices
+            if device.device_id == self._iotty_device.device_id
+        )
+        if isinstance(device, Shutter):
+            self._iotty_device.status = device.status
+            self._iotty_device.percentage = device.percentage
```

Comment:
Why are there 2 different devices?

Reply:
Could you please elaborate?

If you mean two as in one is `self._iotty_device` and two is `device`, well the first one is the entity state, the other is the newly fetched coordinator data

Reply:
No, they're not

Reply:
Why don't we set `self._iotty_device` to `device`?

Reply:
Just wanted to emphasize the fields that are actually changing (i.e. `status` and `percentage) vs the ones that are not changing (e.g. `serial_number`, `device_id`). No problem in assigning directly to the new data from the coordinator if you think that's better

Reply:
I'm not sure what you mean with "map of coordinators". I think you mean something along the lines of using the dictionary? I have changed the two lines here so I directly assign the new device. If you want me to go over this change once more, could you please provide some examples? Or maybe is that something that can be done in a future PR? Seems a bit out of the scope of this one maybe? I have already done the base entity change here, I don't want this PR to become overcomplex